open_samus_returns_rando/samus_returns_patcher.py

```diff
@@ -60,37 +62,86 @@ def _wrap(v: str):
 
     return lua_util.replace_lua_template("custom_init.lua", replacement)
 
-
-def patch_pickups(editor: PatcherEditor, pickups: list):
-    template_bmsad = _read_template_powerup()
+def patch_pickups(editor: PatcherEditor, pickups_config: list[dict]):
+    template_bmsad = editor.get_parsed_asset("actors/items/powerup_chargebeam/charclasses/powerup_chargebeam.bmsad").raw
 
     pkgs_for_lua = set()
 
-    for pickup in pickups:
-        actor_reference = pickup["actor"]
+    for i, pickup in enumerate(pickups_config):
+        LOG.info("Writing pickup %d: %s", i, pickup["item_id"])
+        pkgs_for_level = set(editor.find_pkgs(path_for_level(pickup["pickup_actor"]["scenario"]) + ".bmsld"))
+        pkgs_for_lua.update(pkgs_for_level)
+
+        actor_reference = pickup["pickup_actor"]
         scenario = editor.get_scenario(actor_reference["scenario"])
         actor_name = actor_reference["actor"]
 
-        model_name: str = pickup["type"]
-        pickup_type = get_data(model_name)
-
         found_actor = False
         for actors in scenario.raw.actors:
             if actor_name in actors:
                 actor = actors[actor_name]
-                actor["type"] = pickup["type"]
+                actor["model"] = pickup["model"]
```

Comment:
this line should be further down, next to where you edit `actor.type`. also, you may prefer to write it like this instead: `actor.model = pickup["model"]`. (you can't always use these syntaxes interchangeably! but for most things returned by MEDS you can)

Reply:
actually, on second glance - is this line even supposed to be here at all? you might actually want to just remove it altogether